src/accordion/accordion.ts

```diff
@@ -175,6 +175,44 @@ export class NgbAccordion implements AfterContentChecked {
     }
   }
 
+  /**
+   * Programmatically expand a panel with a given id if it is closed.
+   */
+  expand(panelId: string) {
+    const panel = this.panels.find(p => p.id === panelId);
+
+    if (panel && !panel.disabled && !panel.isOpen) {
+      this.toggle(panelId);
+    }
+  }
+
+  /**
+   * Programmatically collapse a panel with a given id if it is open.
+   */
+  collapse(panelId: string) {
+    const panel = this.panels.find(p => p.id === panelId);
+
+    if (panel && !panel.disabled && panel.isOpen) {
+      this.toggle(panelId);
+    }
+  }
+
+  /**
+   * Programmatically collapse all panels.
+   */
+  collapseAll() {
+    this.panels.forEach(panel => { panel.isOpen = false; });
+    this._updateActiveIds();
+  }
+
+  /**
+   * Returns if a panel is currently expanded.
+   */
+  isExpanded(panelId): boolean {
+    const panel = this.panels.find(p => p.id === panelId);
+    return panel.isOpen;
```

Comment:
This will crash if there is no panel with a given `panelId` if I'm not mistaken? Could you please add a test for this case and fix the code if needed?